src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/TelemetryReporterTests.cs

```diff
@@ -322,4 +323,107 @@ public void TrackLspRequest()
                 Assert.Equal(correlationId, correlationProperty.Value);
             });
     }
+
+    [Fact]
+    public void ReportFault_OperationCanceledExceptionWithoutInnerException_SkipsFaultReport()
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var exception = new OperationCanceledException("OCE", innerException: null);
+
+        // Act
+        reporter.ReportFault(exception, "Test message");
+
+        // Assert
+        Assert.Empty(reporter.Events);
+    }
+
+    [Fact]
+    public void ReportFault_TaskCanceledExceptionWithoutInnerException_SkipsFaultReport()
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var exception = new TaskCanceledException("TCE", innerException: null);
+
+        // Act
+        reporter.ReportFault(exception, "Test message");
+
+        // Assert
+        Assert.Empty(reporter.Events);
+    }
+
+    [Theory]
+    [InlineData(3)]
+    public void ReportFault_InnerExceptionOfOCEIsAnotherOCE_SkipsFaultReport(int depth)
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var innerMostException = new OperationCanceledException();
+        var exception = CreateNestedException(typeof(OperationCanceledException), depth, innerMostException);
```

Comment:
Nit: I would get rid of the helper methods and simplify this. This line could be replace with `new OperationCanceledException("", innerException: new OperationCanceledExcept())` and the test would still be valid.

Even having a nesting depth of 3 was particularly useful, having a theory with one data point, and two helper methods to create the structure, makes it harder to reason about what is going on and what the final shape of the exceptions looks like. Simply writing out the code directly using the constructors would make it much clearer IMO.

Reply:
Good point! thanks for the feedback